### DISCUSSION TITLE
How to check if the device was authenticated?

### DISCUSSION THREAD

**Discussion by LaAlexe** [yume-chan/ya-webadb#615] [Q&A]
I found the method await AdbDaemonTransport.authenticate() during connection to device. During this method I got a prompt on my device 'Allow USB debugging', how to check if it was allowed?  Is there are any possibility to check if device was authenticated or not?

**Comment by yume-chan** [comment]
`authenticate()` returns when user clicks "Allow", or if the private key is already trusted by the device.
